### PR TITLE
Rm xstream provisioning list query and response

### PR DESCRIPTION
**Pull request by erikrozendaal**
Replaced by Java DOM XML implementations.

### CODE REVIEW

File: src/main/java/net/ripe/rpki/commons/provisioning/identity/IdentitySerializer.java [hunk 1]
```diff
@@ -32,77 +32,30 @@
 import net.ripe.rpki.commons.provisioning.x509.ProvisioningIdentityCertificate;
 import net.ripe.rpki.commons.provisioning.x509.ProvisioningIdentityCertificateParser;
 import net.ripe.rpki.commons.validation.ValidationResult;
+import net.ripe.rpki.commons.xml.DomXmlSerializer;
 import org.w3c.dom.Document;
 import org.w3c.dom.Node;
```
Comment (wibisono) [inline src/main/java/net/ripe/rpki/commons/provisioning/identity/IdentitySerializer.java:37]:
```suggestion

```
Idea said this is not used.

File: src/main/java/net/ripe/rpki/commons/provisioning/payload/AbstractProvisioningPayload.java [hunk 2]
```diff
@@ -30,8 +30,9 @@
 package net.ripe.rpki.commons.provisioning.payload;
 
 import com.thoughtworks.xstream.annotations.XStreamAsAttribute;
```
Comment (wibisono) [inline src/main/java/net/ripe/rpki/commons/provisioning/payload/AbstractProvisioningPayload.java:32]: I suspect this annotation also can go.

Comment (erikrozendaal) [reply src/main/java/net/ripe/rpki/commons/provisioning/payload/AbstractProvisioningPayload.java:32]: Once everything is converted we can remove the xstream dependencies.

File: src/main/java/net/ripe/rpki/commons/provisioning/identity/IdentitySerializer.java [hunk 3]
```diff
@@ -32,77 +32,28 @@
...
-        final StringWriter sw = new StringWriter();
-        transformer.transform(new DOMSource(document), new StreamResult(sw));
-
-        return sw.toString();
-    }
-
     protected ProvisioningIdentityCertificate getProvisioningIdentityCertificate(final String bpkiTa) {
         final ProvisioningIdentityCertificateParser parser = new ProvisioningIdentityCertificateParser();
-        parser.parse(ValidationResult.withLocation("unknown.cer"), Base64.getDecoder().decode(bpkiTa));
+        parser.parse(ValidationResult.withLocation("unknown.cer"), Base64.getMimeDecoder().decode(bpkiTa));
```
Comment (ties) [inline src/main/java/net/ripe/rpki/commons/provisioning/identity/IdentitySerializer.java:56]: Is this the one where it is needed to use the mime decoder for the specific object test? If so, I'd like an inline comment about that for the situation where this comment applies

File: src/main/java/net/ripe/rpki/commons/provisioning/payload/common/CertificateElement.java [hunk 4]
```diff
@@ -34,14 +34,14 @@
...
-import org.apache.commons.lang.builder.ToStringBuilder;
+import net.ripe.rpki.commons.util.EqualsSupport;
 
 import java.net.URI;
 import java.util.Iterator;
 import java.util.List;
 
 @XStreamConverter(CertificateElementConverter.class)
-public class CertificateElement {
+public class CertificateElement extends EqualsSupport {
```
Comment (ties) [inline src/main/java/net/ripe/rpki/commons/provisioning/payload/common/CertificateElement.java:44]: For these type of POJO's you would almost want to add lombok

File: src/main/java/net/ripe/rpki/commons/provisioning/payload/AbstractProvisioningPayloadXmlSerializer.java [hunk 5]
```diff
@@ -0,0 +1,148 @@
...
+
+    @Override
+    public T deserialize(String xml) {
+        try (final Reader characterStream = new StringReader(xml)) {
+            Document doc = getDocumentBuilder().parse(new InputSource(characterStream));
+
+            Element message = getElement(doc, "message")
+                    .orElseThrow(() -> new DomXmlSerializerException("message element not found"));
+
+            String versionString = getAttributeValue(message, "version")
```
Comment (ties) [inline src/main/java/net/ripe/rpki/commons/provisioning/payload/AbstractProvisioningPayloadXmlSerializer.java:87]: for consistency, `getRequiredAttributeValue`?

**Review by ties — APPROVED** [review]

File: src/main/java/net/ripe/rpki/commons/provisioning/payload/common/CertificateElement.java (hunk 4 above)
Comment (erikrozendaal) [reply src/main/java/net/ripe/rpki/commons/provisioning/payload/common/CertificateElement.java:44]: Yes, `EqualsSupport` is our slow and hacky solution for not having Lombok at the time. Here it is used to make a nice serialization/deserialization test, so the performance is not a problem.

File: src/main/java/net/ripe/rpki/commons/provisioning/payload/AbstractProvisioningPayloadXmlSerializer.java (hunk 5 above)
Comment (erikrozendaal) [reply src/main/java/net/ripe/rpki/commons/provisioning/payload/AbstractProvisioningPayloadXmlSerializer.java:87]: Done

**Review by wibisono — APPROVED** [review]

File: src/main/java/net/ripe/rpki/commons/provisioning/payload/AbstractProvisioningPayloadXmlSerializer.java [hunk 6]
```diff
@@ -0,0 +1,148 @@
...
+                throw new DomXmlSerializerException("version attribute is not a number: " + versionString, e);
+            }
+            if (!SUPPORTED_VERSION.equals(version)) {
+                throw new DomXmlSerializerException("version attribute is not '1': " + version);
+            }
+
+            String sender = getAttributeValue(message, "sender")
+                    .orElseThrow(() -> new DomXmlSerializerException("sender attribute not found"));
+
+            String recipient = getAttributeValue(message, "recipient")
```
Comment (lolepezy) [inline src/main/java/net/ripe/rpki/commons/provisioning/payload/AbstractProvisioningPayloadXmlSerializer.java:108]: and `getRequiredAttributeValue` here as well?

**Review by lolepezy — APPROVED** [review]

File: src/main/java/net/ripe/rpki/commons/provisioning/payload/common/GenericClassElement.java [hunk 7]
```diff
@@ -132,6 +132,18 @@ public IpResourceSet getResourceSetIpv6() {
         return resourceSetIpv6;
     }
 
+    public void setResourceSetAs(IpResourceSet resourceSetAs) {
```
[the remark this answers is not in the document]
Comment (erikrozendaal) [reply src/main/java/net/ripe/rpki/commons/provisioning/payload/common/GenericClassElement.java:135]: Good idea when https://github.com/mplushnikov/lombok-intellij-plugin/issues/952 is fixed.

File: src/main/java/net/ripe/rpki/commons/provisioning/payload/AbstractProvisioningPayload.java [hunk 8]
```diff
@@ -30,8 +30,9 @@
...
+import net.ripe.rpki.commons.util.EqualsSupport;
 
-public abstract class AbstractProvisioningPayload {
+public abstract class AbstractProvisioningPayload extends EqualsSupport {
```
[the remark this answers is not in the document]
Comment (erikrozendaal) [reply src/main/java/net/ripe/rpki/commons/provisioning/payload/AbstractProvisioningPayload.java:35]: Good idea when mplushnikov/lombok-intellij-plugin#952 is fixed.

File: src/main/java/net/ripe/rpki/commons/provisioning/payload/list/response/ResourceClassListResponsePayloadSerializer.java [hunk 9]
```diff
@@ -0,0 +1,151 @@
...
+    private static final DateTimeConverter DATE_TIME_CONVERTER = new DateTimeConverter();
+    private static final IpResourceSetProvisioningConverter IP_RESOURCE_SET_PROVISIONING_CONVERTER = IpResourceSetProvisioningConverter.INSTANCE;
+    private static final CertificateUrlListConverter CERTIFICATE_URL_LIST_CONVERTER = CertificateUrlListConverter.INSTANCE;
+
+    public ResourceClassListResponsePayloadSerializer() {
+        super(PayloadMessageType.list_response);
+    }
+
+    protected ResourceClassListResponsePayload parseXmlPayload(Element message) {
+        List<ResourceClassListResponseClassElement> classes = new ArrayList<>();
```
[the remark this answers is not in the document]
Comment (erikrozendaal) [reply src/main/java/net/ripe/rpki/commons/provisioning/payload/list/response/ResourceClassListResponsePayloadSerializer.java:67]: Is fixed in a later PR.

**Pull request merged**